src/tracking_state.jl

```diff
@@ -63,9 +106,9 @@ struct TrackingState{
         CALF <: AbstractLoopFilter,
         COLF <: AbstractLoopFilter,
         CN <: AbstractCN0Estimator,
-        DS <: DownconvertedSignalCPU, # Union{DownconvertedSignalCPU, CuArray{ComplexF32}}
-        CAR <: CarrierReplicaCPU, # Union{CarrierReplicaCPU, CuArray{ComplexF32}}
-        COR <: Vector{Int8}, # Union{Vector{Int8}, CuArray{Float32}}
+        DS <: Union{DownconvertedSignalCPU, DownconvertedSignalGPU},
+        CAR <: Union{CarrierReplicaCPU, CarrierReplicaGPU},
```

Comment:
I think you can leave `CarrierReplicaCPU` even for the GPU case and initialize that as an empty array.